src/atomicvar.h

```diff
@@ -62,13 +62,61 @@
 #ifndef __ATOMIC_VAR_H
 #define __ATOMIC_VAR_H
 
+/* Define redisAtomic for atomic variable. */
+#define redisAtomic
+
 /* To test Redis with Helgrind (a Valgrind tool) it is useful to define
  * the following macro, so that __sync macros are used: those can be detected
  * by Helgrind (even if they are less efficient) so that no false positive
  * is reported. */
 // #define __ATOMIC_VAR_FORCE_SYNC_MACROS
 
-#if !defined(__ATOMIC_VAR_FORCE_SYNC_MACROS) && defined(__ATOMIC_RELAXED) && !defined(__sun) && (!defined(__clang__) || !defined(__APPLE__) || __apple_build_version__ > 4210057)
+/* There will be many false positives if we test Redis with Helgrind, since
+ * Helgrind can't understand we have imposed ordering on the program, so
+ * we use macros in helgrind.h to tell Helgrind inter-thread happens-before
+ * relationship explicitly for avoiding false positives.
+ *
+ * For more details, please see: valgrind/helgrind.h and
+ * https://www.valgrind.org/docs/manual/hg-manual.html#hg-manual.effective-use
+ *
+ * These macros take effect only when 'make helgrind', and you must first
+ * install Valgrind in the default path configuration. */
+#ifdef __ATOMIC_VAR_FORCE_SYNC_MACROS
+#include <valgrind/helgrind.h>
+#define REDIS_ANNOTATE_HAPPENS_BEFORE(v) ANNOTATE_HAPPENS_BEFORE(v)
+#define REDIS_ANNOTATE_HAPPENS_AFTER(v)  ANNOTATE_HAPPENS_AFTER(v)
+#else
+#define REDIS_ANNOTATE_HAPPENS_BEFORE(v) ((void) v)
+#define REDIS_ANNOTATE_HAPPENS_AFTER(v)  ((void) v)
+#endif
```

Comment:
Can we simplify this and drop the REDIS annotation?
```
#ifdef __ATOMIC_VAR_FORCE_SYNC_MACROS
#include <valgrind/helgrind.h>
#else
#define ANNOTATE_HAPPENS_BEFORE(v) ((void) v)
#define ANNOTATE_HAPPENS_AFTER(v)  ((void) v)
#endif

Reply:
I think that's ok

Reply:
done